dietpi/dietpi-software

```diff
@@ -8220,7 +8228,71 @@ _EOF_
 		#NextCloud
 		INSTALLING_INDEX=114
 		if (( ${aSOFTWARE_INSTALL_STATE[$INSTALLING_INDEX]} == 1 )); then
+		
+			#If Apache webserver is used, make sure all necessary Apache modules are enabled: https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html#additional-apache-configurations
+			if (( ${aSOFTWARE_INSTALL_STATE[83]} >= 1 )); then
+				a2enmod rewrite headers env dir mime
+			fi
+			
+			#Make sure all necessary PHP modules are enabled: https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation
+			if (( $DISTRO == 3 )); then
+				php5enmod ctype curl dom fileinfo gd iconv intl json mbstring posix simplexml xmlwriter xmlreader zip pdo_mysql exif opcache apcu
+			else
+				phpenmod ctype curl dom fileinfo gd iconv intl json mbstring posix simplexml xmlwriter xmlreader zip pdo_mysql exif opcache apcu
+			fi
+			
+			#Create Nextcloud specific webserver + PHP config.
+			#Apache: https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html#apache-web-server-configuration
+			if (( ${aSOFTWARE_INSTALL_STATE[83]} >= 1 )); then
+				cat << _EOF_ > /etc/apache2/sites-available/nextcloud.conf
```

Comment:
Should we add `nextcloud.conf` to `dietpi/conf` and just copy it here? This would keep the already large grown `dietpi-software` a bid smaller. I already have issues when trying to watch it with mobile 😆.

Reply:
@MichaIng 

> Should we add nextcloud.conf to dietpi/conf/ and just copy it here?

The original reason for keeping everything in ```dietpi-software```, reduces space of RAMdisk usage ```/DietPi``` as each file (regardless of size) uses at least 4K, where as its probably smaller integrated into ```dietpi-software```.

Reply:
Yeah I see, for optimization reasons, also in case of RAM usage, we should keep all config files < 4K inside of dietpi-software.

The online question then is: Why do we keep the the conf/ actually inside ramdisk? Would be a separate project to switch all of them together with all necessary path changes, but on the long term we could choose another folder to put the config files inside, that is outside of ramdisk, e.g. dietpi_userdata. This would then reduce RAM usage by the amount, dietpi-software becomes slimmer 🙂.